test/e2e_node/swap_test.go

```diff
@@ -38,49 +45,229 @@ import (
+				gomega.Expect(nodeUsedMemory.IsZero()).To(gomega.BeFalseBecause("node used memory is zero"))
+
+				swapCapacity = getSwapCapacity(f, sleepingPod)
+				gomega.Expect(swapCapacity.IsZero()).To(gomega.BeFalseBecause("node swap capacity is zero"))
+
+				err := podClient.Delete(context.Background(), sleepingPod.Name, metav1.DeleteOptions{})
+				framework.ExpectNoError(err)
+
+				stressMemAllocSize = multiplyQuantity(divideQuantity(nodeTotalMemory, 1000), 4)
+
+				ginkgo.By(fmt.Sprintf("Setting node values. nodeName: %s, nodeTotalMemory: %s, nodeUsedMemory: %s, swapCapacity: %s, stressMemAllocSize: %s",
+					nodeName, nodeTotalMemory.String(), nodeUsedMemory.String(), swapCapacity.String(), stressMemAllocSize.String()))
+			})
+
+			ginkgo.Context("LimitedSwap", func() {
+				tempSetCurrentKubeletConfig(f, enableLimitedSwap)
+
+				getRequestBySwapLimit := func(swapPercentage int64) *resource.Quantity {
+					gomega.ExpectWithOffset(1, swapPercentage).To(gomega.And(
+						gomega.BeNumerically(">=", 1),
+						gomega.BeNumerically("<=", 100),
+					), "percentage has to be between 1 and 100")
+
+					// <swap-limit> = <memory-request>*(<swap-capacity>/<node-capacity>).
+					// if x is the percentage, and <memory-request> == (x/100)*<node-capacity>, then:
+					// <swap-limit> = (x/100)*<node-capacity>*(<swap-capacity>/<node-capacity>) = (x/100)*<swap-capacity>.
+					return multiplyQuantity(divideQuantity(nodeTotalMemory, 100), swapPercentage)
+				}
+
+				getStressPod := func(stressSize *resource.Quantity) *v1.Pod {
+					pod := getStressPod(f, stressSize, stressMemAllocSize)
+					pod.Spec.NodeName = nodeName
+
+					return pod
+				}
+
+				ginkgo.It("should be able over-commit the node memory", func() {
+					stressSize := cloneQuantity(nodeTotalMemory)
+
+					stressPod := getStressPod(stressSize)
+					// Request will use a lot more swap memory than needed, since we don't test swap limits in this test
+					memRequest := getRequestBySwapLimit(30)
+					setPodMmoryResources(stressPod, memRequest, noLimits)
+					gomega.Expect(qos.GetPodQOS(stressPod)).To(gomega.Equal(v1.PodQOSBurstable))
+
+					ginkgo.By(fmt.Sprintf("creating a stress pod with stress size %s and request of %s", stressSize.String(), memRequest.String()))
+					stressPod = runPodAndWaitUntilScheduled(f, stressPod)
+
+					ginkgo.By("Expecting the swap usage to be non-zero")
+					var swapUsage *resource.Quantity
+					gomega.Eventually(func() error {
+						stressPod = getUpdatedPod(f, stressPod)
+						gomega.Expect(stressPod.Status.Phase).To(gomega.Equal(v1.PodRunning), "pod should be running")
+
+						var err error
+						swapUsage, err = getSwapUsage(f, stressPod)
+						if err != nil {
+							return err
+						}
+
+						if swapUsage.IsZero() {
+							return fmt.Errorf("swap usage is zero")
+						}
+
+						// Better to delete the stress pod ASAP to avoid node failures
+						_ = podClient.Delete(context.Background(), stressPod.Name, metav1.DeleteOptions{})
```

Comment:
> I don't think we should call this more than once?
> Could we move it out of the eventually?

It is being called only once before the Eventually clause returns no err (nil).
I basically wanted to delete the pod ASAP to avoid compromising the node, but there's no reason to do so right after the eventually clause ends.

Thanks @kannon92! I agree, the code is much clearer this way.
Done.

> And can we assert there is no error?

Sure. Done.